### DISCUSSION TITLE
CRUD returns Bad Request on $unset PATCH

### DISCUSSION THREAD

**Discussion by calairoh** [mia-platform/community#177] [Q&A] [answered]
Hi,

We're experimenting some problems with the `$unset` operation of the CRUD Patch. We have the following request:
```curl
curl --request PATCH \
  --url http://localhost:3000/collection \
  --header 'Content-Type: application/json' \
  --data '{ "$unset": { "name": true } }'
```
And we get the following error:
```json
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "body.$unset should NOT have additional properties"
}
```

The field `name` exists in the collection. Indeed, we are able to perform `$set` operations.

Are we performing the request in the wrong way?

**Comment by edocrippaofficial** [comment]


**Reply by edocrippaofficial** [reply]
Nope, it worked even without the ending slash

**Reply by calairoh** [reply, marked as answer]
Actually, we solved the issue. The problem was that the `name` field was set to `required` in the MongoDB CRUD section.

It's our trivial mistake, however the error message returned by the CRUD is not immediately understandable.

**Reply by edocrippaofficial** [reply]
You could also find that by requesting the swagger from http://localhost:3000/documentation/json  
From there you could clearly see that the unset schema contains less fields than the $set

**Reply by danibix95** [reply]
> Actually, we solved the issue. The problem was that the `name` field was set to `required` in the MongoDB CRUD section.
> 
> It's our trivial mistake, however the error message returned by the CRUD is not immediately understandable.

Thanks for the feedback @calairoh
FYI @nicocampagna
